### ISSUE TITLE
Cannot upload io.StringIO with non-ASCII characters

### ISSUE THREAD

**Issue by ilia1243** [Netcracker/KubeMarine#342]
Steps to reproduce:
1. Create template files/test_cm.yaml.j2
```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-cm
data:
  cluster_name: "{{ cluster_name }}"
  unicode_symbol: "😀"
```
2. Create custom plugin in inventory
```yaml
plugins:
  ...
  my_plugin:
    install: true
    installation:
      procedures:
        - template:
            source: files/test_cm.yaml.j2
```
3. Run `install --tasks deploy.plugins`

AR: fails with `size mismatch in put!  121 != 118`, see https://github.com/paramiko/paramiko/issues/1133

The problem is likely actual for any upload of io.StringIO instance.

Possible solution:

When uploading of any text, manually encode it with 'utf-8' and upload as io.BytesIO

If the text is obtained from file on Windows deployer, it will need to decode it first with the same 'utf-8' to not lose the utf-8 symbols.

In other words, it is necessary to introduce new requirement for encoding of input text files and **always open for reading and writing with 'utf-8'** and always upload as io.BytesIO.

Prototype: https://github.com/Netcracker/KubeMarine/commit/eedfbf916b999b7eeb343d58a34d0c725e8bcc14
